crates/sui-indexer-alt/src/pipeline/sequential/committer.rs

```diff
@@ -108,24 +112,56 @@ pub(super) fn committer<H: Handler + 'static>(
                         continue;
                     };
 
+                    // Determine whether we need to hold back checkpoints from being committed
+                    // because of checkpoint lag.
+                    //
+                    // TODO(amnn): Test this (depends on migrations and tempdb)
+                    let commit_hi = match (checkpoint_lag, pending.last_key_value()) {
+                        (Some(lag), None) => {
+                            debug!(pipeline = H::NAME, lag, "No pending checkpoints");
+                            if rx.is_closed() && rx.is_empty() {
+                                break;
+                            } else {
+                                continue;
+                            }
+                        }
+
+                        (Some(lag), Some((pending_hi, _))) if *pending_hi < lag => {
+                            debug!(pipeline = H::NAME, lag, pending_hi, "Priming pipeline");
+                            if rx.is_closed() && rx.is_empty() {
+                                break;
+                            } else {
+                                continue;
+                            }
+                        }
+
+                        (Some(lag), Some((pending_hi, _))) => Some(*pending_hi - lag),
```

Comment:
it's prob okay while one difference between the delay here and the delay in production today is that in production, the delay is based on the committed watermark instead of the pending highest checkpoint to commit; the behavior will be roughly the same when updated b/c pending will be mostly empty, and when `pending` has a populated queue everywhere like when backfill, the checkpoint diff between `objects` and `objects_snapshot` will be smaller than the configured lag.